lib/Optimizer/Transforms/LiftArrayAlloc.cpp

```diff
@@ -28,6 +28,9 @@ namespace cudaq::opt {
 using namespace mlir;
 
 namespace {
+
+static constexpr const char ReadOnlySuffix[] = ".rodata";
```

Comment:
This change to omit any uniquing will result in collisions in the symbol namespace as it is entirely possible to have more than 1 array being constructed in a kernel.